src/client/Client.cc

```diff
@@ -11403,6 +11403,8 @@ void Client::C_Write_Finisher::finish_io(int r)
 {
   bool fini;
 
+  std::scoped_lock lock(clnt->client_lock);
```

Comment:
`_write()` can invoke this callback with client_lock held and that could cause a deadlock. See:

```    if (onfinish) {
      // handle non-blocking caller (onfinish != nullptr), we can now safely
      // release all the managed pointers, but we might need to do something
      // with iofinisher.
      Context *iof = iofinish.release();
      C_Write_Finisher *cwfp = cwf.release();

      if (r < 0) {
        // should not get here, but...
        iof->complete(r); <<<< here
      }
```

Reply:
Oh I had missed the error cases. Done